tutorials/intermediate/constrainedlayout_guide.py

```diff
@@ -430,6 +450,34 @@ def docomplicated(suptitle=None):
 example_plot(axs[0], fontsize=12)
 axs[1].set_position([0.2, 0.2, 0.4, 0.4])
 
+###############################################################################
+# .. _compressed_layout:
+#
+# Grids of fixed aspect-ratio Axes: "compressed" layout
+# =====================================================
+#
+# ``constrained_layout`` operates on the grid of "original" positions for
+# axes. However, when Axes have fixed aspect ratios, one side is usually made
+# shorter, and leaves large gaps in the shortened direction. In the following,
+# the Axes are square, but the figure quite wide so there is a horizontal gap:
+
+fig, axs = plt.subplots(2, 2, figsize=(5, 3),
+                        sharex=True, sharey=True, layout='constrained')
+for ax in axs.flat:
+    ax.imshow(arr)
+fig.suptitle("fixed-aspect plots, layout='constrained'")
+
+# One obvious way of fixing this is to make the figure size more square,
+# however, closing the gaps exactly requires trial and error.  For simple grids
+# of Axes we can use ``layout="compressed"`` to do the job for us:
+
+fig, axs = plt.subplots(2, 2, figsize=(5, 3),
+                        sharex=True, sharey=True, layout='compressed')
+for ax in axs.flat:
+    ax.imshow(arr)
+fig.suptitle("fixed-aspect plots, layout='compressed'")
```

Comment:
This seems to look the same as the constrained version?

Reply:
Agreed, https://output.circle-artifacts.com/output/job/4eee1b1a-1268-4881-8d21-61000c011c8e/artifacts/0/doc/build/html/tutorials/intermediate/constrainedlayout_guide.html#grids-of-fixed-aspect-ratio-axes-compressed-layout

but I have no idea what is happening here.  This example works fine manually, just using Agg, etc.  I'm flummoxed why the sphinx version doesn't work...

Reply:
Fixed in newest version - there was some weird interactions between the rcParam and subsequent calls.  maybe worth chasing down, but I could not reproduce without sphinx-gallery, so I guess sphinx-galery is using some weird rcParam contexts?

Reply:
See https://output.circle-artifacts.com/output/job/a187bd70-81b0-4ede-ae1c-ab250a32d50c/artifacts/0/doc/build/html/tutorials/intermediate/constrainedlayout_guide.html#grids-of-fixed-aspect-ratio-axes-compressed-layout

Reply:
I think the context remains the same through a single file.

Reply:
Yes, but even if I change the rcParam in a normal script, and then specify the kwarg, `compress` works fine.  But in sphinx- gallery it has this weird problem.